crates/rpc/rpc-engine-api/src/engine_api.rs

```diff
@@ -677,13 +677,17 @@ where
     /// Caution: This should not accept the `withdrawals` field
     async fn new_payload_v1(&self, payload: ExecutionPayloadV1) -> RpcResult<PayloadStatus> {
         trace!(target: "rpc::engine", "Serving engine_newPayloadV1");
+        Ok(Self::new_payload_v1_metered(self, payload).await?)
+    }
+
+    async fn new_payload_v1_metered(
+        &self,
+        payload: ExecutionPayloadV1,
+    ) -> RpcResult<PayloadStatus> {
         let start = Instant::now();
-        let gas_used = payload.gas_used;
-        let res = Self::new_payload_v1(self, payload).await;
-        let elapsed = start.elapsed();
-        self.inner.metrics.latency.new_payload_v1.record(elapsed);
-        self.inner.metrics.new_payload_response.update_response_metrics(&res, gas_used, elapsed);
-        Ok(res?)
+        let res = Self::fork_choice_updated_v1(self, fork_choice_state, payload_attributes).await;
+        self.inner.metrics.latency.fork_choice_updated_v1.record(start.elapsed());
+        self.inner.metrics.fcu_response.update_response_metrics(&res);
```

Comment:
this must be moved next to

https://github.com/paradigmxyz/reth/blob/d31e1d601d2f97bdb6b7c7f39829456e6865fe43/crates/rpc/rpc-engine-api/src/engine_api.rs#L135-L135

Reply:
Got it, now is not throwing any errors. But it was weird because I was doing that thing and I was getting errors. Anyways, thanks.